L1Trigger/L1TNtuples/plugins/L1UpgradeTfMuonTreeProducer.cc

```diff
@@ -117,18 +138,42 @@ L1UpgradeTfMuonTreeProducer::~L1UpgradeTfMuonTreeProducer() {}
 
 // ------------ method called to for each event  ------------
 void L1UpgradeTfMuonTreeProducer::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  edm::EDGetTokenT<l1t::RegionalMuonCandBxCollection> legacybmtfMuonToken;
+  edm::EDGetTokenT<l1t::RegionalMuonCandBxCollection> kbmtfMuonToken;
+  if (isEMU_) {
+    legacybmtfMuonToken = bmtfMuonToken_;
+    kbmtfMuonToken = bmtf2MuonToken_;
+  } else {
+    iEvent.getByToken(fedToken_, feds_);
+    // Get fw version
+    unsigned algoFwVersion { getAlgoFwVersion() };
+    if (algoFwVersion < 2499805536) {  //95000160(hex)
+      // Legacy was triggering (and therefore in the main collection)
+      legacybmtfMuonToken = bmtfMuonToken_;
+      kbmtfMuonToken = bmtf2MuonToken_;
+    } else {
+      // kBMTF was triggering
+      legacybmtfMuonToken = bmtf2MuonToken_;
+      kbmtfMuonToken = bmtfMuonToken_;
+    }
+  }
```

Comment:
Hi @dinyar, all the changes make sense to me - only one detail here
We probably need to make this part of the code era dependent, for Run2 old-bmtf is coll 1 and for Run3 kalman-bmtf is coll 1

Reply:
Hi Pano,

I thought if it's dependent on the firmware version we're safe. The reason I needed to use the firmware version was that we did take data in Run-2 with kBMTF triggering, so if make it only era dependent we run into problems there. I think this was actually a requirement because Ioanna was at the time doing her studies on the Run-2 data with kBMTF.

Reply:
you are right about this, didnt thought of this - thx seems fine
FYI @ckamtsikis